### ISSUE TITLE
cURL error 35: error:1414D172:SSL routines:tls12_check_peer_sigalg:wrong signature type

### ISSUE THREAD

**Issue by chemical1979** [FreshRSS/FreshRSS#3029]
I'm not sure whether this can be fixed by some changed command line arguments, or where to look in general.
I am running Raspbian Buster and master branch of FreshRSS. I believe either since upgrading to buster or upgrading FreshRSS I did not receive new posts in a specific feed. I noticed very late.

Anyway, this is the Feed URL that was working before:

https://www.presseportal.de/rss/dienststelle_55625.rss2

And this is the error that shows up in the logs: 
`URL error 35: error:1414D172:SSL routines:tls12_check_peer_sigalg:wrong signature type`

In a similar bug report for [debian](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=934453) and [here ](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=912759) they said that the web service was faulty, they are only offering signature types that the client do not support. I guess I can not change this for pressportal.de, is there any way around it to make it work?

[SSLlabs ](https://www.ssllabs.com/ssltest/analyze.html?d=www.presseportal.de) show their protocol support is zero which leads to a "B" rating, so it is probably a misconfiguration on their side, though it seems that all browsers ignore this, but only curl fails. Anything I can do on my end?

**Comment by aledeg** [comment]
I ran your URL in the w3c validator and it has the same error. My guess is that the server has an error.
https://validator.w3.org/feed/check.cgi?url=presseportal.de%2Frss%2Fdienststelle_55625.rss2

**Comment by Frenzie** [comment]
It works if you put `SECLEVEL 1` in /etc/ssl/openssl.cnf to allow for insecure outdated signatures. But again, that's insecure and outdated.

More info [here](https://wiki.debian.org/ContinuousIntegration/TriagingTips/openssl-1.1.1).

> In Debian the defaults are set to more secure values by default. This is done in the /etc/ssl/openssl.cnf config file. At the end of the file there is:
>
>    [system_default_sect]
>    MinProtocol = TLSv1.2
>    CipherString = DEFAULT@SECLEVEL=2
>
> This can results in errors such as:
>
>    dh key too small
>    ee key too small
>    ca md too weak
>
>This is caused by the SECLEVEL 2 setting the security level to 112 bit. This means that RSA and DHE keys need to be at least 2048 bit long. SHA-1 is no longer supported for signatures in certificates and you need at least SHA-256. Note that CAs have stopped issuing certificates that didn't meet those requirements in January 2015, and since January 2017 all valid CA certificates should meet those requirements. However there are certificates generated by private CAs or that are in a test suite that do not meet those requirements.
>
>SECLEVEL 1 was the default in previous versions and is at the 80 bit security level, requiring a 1024 bit RSA key.

**Comment by chemical1979** [comment]
`curl -X GET https://www.presseportal.de`

* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (OUT), TLS alert, handshake failure (552):
* error:1414D172:SSL routines:tls12_check_peer_sigalg:wrong signature type
* Closing connection 0
curl: (35) error:1414D172:SSL routines:tls12_check_peer_sigalg:wrong signature type


Yes, they seem to ignore the list of hashing algorithms that cURL offers and their reply is signed with one of those missing in our offer. At least that is what I understand is the root issue. I'm just wondering if there is any way to make cURL more tolerant (which probably basically means unsecure).

**Comment by chemical1979** [comment]
> It works if you put `SECLEVEL 1` in /etc/ssl/openssl.cnf to allow for insecure outdated signatures. But again, that's insecure and outdated.

That seems to be a valid workaround for me. Thank you!
I'll try to reach out to presseportal.de to see whether they will change things, but I doubt it.

**Comment by chemical1979** [comment]
I closed too fast. This works for CLI curl, but not for PHP-curl

**Comment by Frenzie** [comment]
@chemical1979 I think that probably just means you didn't restart your server.

**Comment by Frenzie** [comment]
Or more to the point your Apache/Nginx/whatever you're using.

**Comment by Alkarex** [comment]
See also the SSL feed option in FreshRSS:
![image](https://user-images.githubusercontent.com/1008324/83605855-f39c8080-a578-11ea-82cc-586a9bc9f723.png)

**Comment by chemical1979** [comment]
I tried reverting the change in openssl and checked the feed SSL verification option in FreshRSS, but that did not help.

So current workaround is to set `CipherString = DEFAULT@SECLEVEL=2` from 2 to 1 in `/etc/ssl/openssl.cnf`

**Comment by james5294** [comment]
i have this same issue.. im a little confused how to fix it. can someone help me with an eli5 answer =)

**Comment by Alkarex** [comment]
Instead of manually generating certificates, I recommend using Let's Encrypt.
If you manually generate certificates, you are supposed to deploy them to your clients.

Otherwise, at feed level, see the advanced options
![image](https://user-images.githubusercontent.com/1008324/140606488-ef0fe07b-b1ef-4eee-a1e5-ee6df2598db2.png)
(We could add more SSL options if needed)

And at global level, see the `'curl_options'` section of your `./data/config.php` file and the corresponding options https://php.net/function.curl-setopt
